src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -804,10 +804,13 @@ private BoundExpression BindTupleExpression(TupleExpressionSyntax node, Diagnost
                 }
                 else
                 {
-                    inferredName = InferTupleElementName(argumentSyntax.Expression);
+                    if (this.Compilation.LanguageVersion >= LanguageVersion.CSharp7_1)
```

Comment:
This check occurs several times in `Binder`. Consider extracting a helper method.